components/layout/block.rs

```diff
@@ -1701,8 +1701,10 @@ impl BlockFlow {
             Display::Table |
             Display::InlineBlock |
             Display::Flex => FormattingContextType::Other,
-            _ if style.get_box().overflow_x != StyleOverflow::Visible ||
-                style.get_box().overflow_y != StyleOverflow::Visible ||
+            _ if (style.get_box().overflow_x != StyleOverflow::Visible &&
+                style.get_box().overflow_x != StyleOverflow::Clip) ||
+                (style.get_box().overflow_y != StyleOverflow::Visible &&
+                    style.get_box().overflow_y != StyleOverflow::Clip) ||
```

Comment:
This is legacy layout, so I wouldn't bother with `overflow: clip` here.

But if you are changing it, then at least use `is_scrollable()`:

```suggestion
            _ if style.get_box().overflow_x.is_scrollable() ||
                style.get_box().overflow_y.is_scrollable() ||
```

Or in fact both axes must have the same scrollability, so just checking one should suffice.